### PR TITLE
Feature/85-build-the-component-CreateActivityButton

### PR DESCRIPTION
**Pull request by rusisolanki**
- Built the CreateActivityButton component
- Created CreateActivityButtonStyles.css file for custom CSS
- Built the CheckIcon component
- Made one change in the ButtonStyles.css file

### CODE REVIEW

**Comment by erenfn** [comment]
You didn't commit your changes.

Also open pull request to develop, not to master

**Review by erenfn — CHANGES_REQUESTED** [review]
You didn't commit your changes.

Also open pull request to develop, not to master

**Comment by rusisolanki** [comment]
Committed the changes

**Comment by erenfn** [comment]
Check commits. The only commit you pushed is "[Merge remote-tracking branch 'origin/develop' into feature/85-build-t…]" which is a week ago

File: package.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (erenfn) [inline package.json:1]: Why do you have these outside the frontend folder?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (erenfn) [inline .gitignore:1]: Why do you have these outside the frontend folder?

File: package-lock.json
Comment (erenfn) [inline package-lock.json:1]: remove gitignore package json and package lock json

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import Button from "../Button/Button";
+import "./CreateActivityButtonStyles.css";
+import CheckIcon from "../CheckIcon/CheckIcon";
+
+const ActivityButton = () => {
+
+  const listItems = [
```
Comment (erenfn) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:8]: Make list items a prop

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "Announce a new feature",
+    "Publish release notes",
+    "Give more information about a recent news",
+  ];
+
+  return (
+    <div className="bannerStyle">
+      <div className="bannerContents">
+        <h2 className="heading">Use banners to</h2>
```
Comment (erenfn) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:17]: make the heading a prop

File: frontend/src/components/Button/ButtonStyles.css [hunk 5]
```diff
@@ -3,7 +3,7 @@
   background-color: var(--main-purple);
   text-transform: none;
   box-shadow: none;
-  width: 80px;
+  /* width: 80px; */
```
Comment (erenfn) [inline frontend/src/components/Button/ButtonStyles.css:6]: "Can you change the width of the 'Button' without deleting this line? If not, can you modify Button.jsx to allow changing the style?"

**Review by erenfn — DISMISSED** [review]

**Comment by erenfn** [comment]
you can create createActivityButtonData.js under data folder for listitems and heading for testing purposes

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import CheckCircleOutlinedIcon from "@mui/icons-material/CheckCircleOutlined";
+
+export default function CheckIcon({ size, outline, color }) {
```
Comment (swoopertr) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:6]: Please use the same pattern when created a components: 
this : export default function CheckIcon()
or this : const CreateActivityButton()

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import Button from "../Button/Button";
+import "./CreateActivityButtonStyles.css";
+import CheckIcon from "../CheckIcon/CheckIcon";
+import PropTypes from "prop-types";
+import { list } from "../../data/CreateActivityButtonData";
+
+const CreateActivityButton = ({ index }) => {
```
Comment (swoopertr) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:8]: Please use the same pattern when created a components: 
this : export default function CheckIcon()
or this : const CreateActivityButton()

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+  );
+}
+
+CheckIcon.propTypes = {
+  size: PropTypes.oneOf(["small", "medium", "large"]),
+  outline: PropTypes.oneOf([true, false]),
+  color: PropTypes.oneOf(["#7F56D9", "#475467", "#079455"]),
+};
+
+CheckIcon.defaultProps = {
```
Comment (uparkalau) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:24]: do not use defaultProps use default javascript parameters

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+  size: "medium",
+  outline: true,
+  color: "#7F56D9",
```
Comment (uparkalau) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:27]: use constant for colors

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        <CheckCircleIcon sx={{ color: { color } }} />
+      )}
+    </div>
+  );
+}
+
+CheckIcon.propTypes = {
+  size: PropTypes.oneOf(["small", "medium", "large"]),
+  outline: PropTypes.oneOf([true, false]),
+  color: PropTypes.oneOf(["#7F56D9", "#475467", "#079455"]),
```
Comment (uparkalau) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:21]: use constant for colors

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+              <CheckIcon size='small' outline= {true} color='#7F56D9'/>
+              <li className="listText" key={index}>
+                {listItem}
+              </li>
+            </div>
+            ))}
+          </ul>
+      </div>
+      <div className="button">
+        <Button text="Create a banner" className="button-primary-custom buttonStyle" />
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:25]: it has to be parameter for button text.

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+export default function CreateActivityButton({ index }){
+  return (
+    <div className="bannerStyle">
+      <div className="bannerContents">
+          <h2 className="heading">{list[index].heading}</h2>
+          <ul>
+            {list[index].listItem.map((listItem, index) => (
+              <div className="list" key={index}>
+              <CheckIcon size='small' outline= {true} color='#7F56D9'/>
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:16]: constant for colors

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+      </div>
+      <div className="button">
+        <Button text="Create a banner" className="button-primary-custom buttonStyle" />
+      </div>
+    </div>
+  );
+};
+
+CreateActivityButton.propTypes = {
+  index: PropTypes.oneOf([0, 1, 2, 3, 4])
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:32]: what is 0-4 ? use variables to understand what is it.

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+} 
+
+CreateActivityButton.defaultProps = {
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:35]: delete default props

File: frontend/src/components/CheckIcon/CheckIcon.jsx (hunk 6 above)
Comment (uparkalau) [reply frontend/src/components/CheckIcon/CheckIcon.jsx:6]: Yes, please  use const CreateActivityButton() and arrow function.

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 15]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import Button from "../Button/Button";
+import "./CreateActivityButtonStyles.css";
+import CheckIcon from "../CheckIcon/CheckIcon";
+import PropTypes from "prop-types";
+import { list } from "../../data/CreateActivityButtonData";
+
+export default function CreateActivityButton({ index }){
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:8]: const and arrow function

File: frontend/src/components/Button/ButtonStyles.css (hunk 5 above)
Comment (uparkalau) [reply frontend/src/components/Button/ButtonStyles.css:6]: resolve it as well

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <div className="bannerStyle">
+      <div className="bannerContents">
+          <h2 className="heading">{list[index].heading}</h2>
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:12]: Please refactor it we do not need list and index here

File: frontend/src/App.jsx [hunk 17]
```diff
@@ -6,6 +6,7 @@ import PasswordResetPage from "./scenes/login/PassswordResetPage";
 import ForgotPasswordPage from "./scenes/login/ForgotPasswordPage";
 import CheckYourEmailPage from "./scenes/login/CheckYourEmailPage";
 import SetNewPasswordPage from "./scenes/login/SetNewPassword";
+import CreateActivityButton from "./components/CreateActivityButton/CreateActivityButton";
```
Comment (uparkalau) [inline frontend/src/App.jsx:9]: Delete it if you don't use it

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 18]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import CheckCircleOutlinedIcon from "@mui/icons-material/CheckCircleOutlined";
+
+const purple = "#7F56D9"
+const black = '#475467'
```
Comment (uparkalau) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:7]: We have a global constant in a project.

**Comment by rusisolanki** [comment]
> Check commits. The only commit you pushed is "[Merge remote-tracking branch 'origin/develop' into feature/85-build-t…]" which is a week ago

Done

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx (hunk 12 above)
Comment (uparkalau) [reply frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:19]: you still have color='#7F56D9'/> use colors constants that we use globally on the project

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx (hunk 11 above)
Comment (uparkalau) [reply frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:28]: <Button text="Create a banner" how to use it for other pages Tour and Popup and any other pages?

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 19]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import Button from "../Button/Button";
+import "./CreateActivityButtonStyles.css";
+import CheckIcon from "../CheckIcon/CheckIcon";
+import PropTypes from "prop-types";
+import { list } from "../../data/CreateActivityButtonData";
+
+const [hints, popups, banners, helperLinks, tours] = [0, 1, 2, 3, 4]
```
Comment (uparkalau) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:8]: If you would like to use object use this style: "
const CreateActivity = ({ type, onClick }) => {
  const { heading, listItem, buttonText } = activityData[type];
"
Object example:

export const ACTIVITY_TYPES = {
  HINTS: 'hints',
  POPUPS: 'popups',
};

export const activityData = {
  [ACTIVITY_TYPES.HINTS]: {
    heading: 'Use hints to',
    listItem: [
      'Help user understand a complex situation',
      'Give contextual tips and tricks',
      'Give instant help',
    ],
    buttonText: 'Create a hint',
  },

CreateActivity.propTypes = {
  type: PropTypes.oneOf(Object.values(ACTIVITY_TYPES)).isRequired,
  onClick: PropTypes.func.isRequired,
};

<CreateActivity type={ACTIVITY_TYPES.HINTS} onClick={handleCreateHintClick} />

**Review by uparkalau — CHANGES_REQUESTED** [review]
this is my expectation with object implementation:
```

import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button/Button';
import CheckIcon from '../CheckIcon/CheckIcon';
import { ACTIVITY_TYPES, activityData } from '../../data/activityData';
import './CreateActivityStyles.css';

const CreateActivity = ({ type, onClick }) => {
  const { heading, listItem, buttonText } = activityData[type];

  return (
    <div className="create-activity-container">
      <div className="activity-contents">
        <h2 className="activity-heading">{heading}</h2>
        <ul className="activity-list">
          {listItem.map((item, idx) => (
            <li className="activity-list-item" key={idx}>
              <CheckIcon size="small" outline color="var(--primary-color)" />
              <span>{item}</span>
            </li>
          ))}
        </ul>
      </div>
      <Button text={buttonText} className="button-primary-custom activity-button" onClick={onClick} />
    </div>
  );
};

CreateActivity.propTypes = {
  type: PropTypes.oneOf(Object.values(ACTIVITY_TYPES)).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default CreateActivity;
```

File: frontend/src/utils/constants.js [hunk 20]
```diff
@@ -5,3 +5,6 @@ export const API_BASE_URL = 'http://localhost:3000/api';
 export const APP_TITLE = 'Bluewave Onboarding';
 export const SUPPORT_EMAIL = 'support@bluewave.com';
 
+export const purple = "#7F56D9"
```
Comment (uparkalau) [inline frontend/src/utils/constants.js:8]: If you have any global colors use it from here or add it there.  src/styles/variables.css

**Review by uparkalau — CHANGES_REQUESTED** [review]

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 21]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import PropTypes from "prop-types";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import CheckCircleOutlinedIcon from "@mui/icons-material/CheckCircleOutlined";
+import { purple, black, green } from "../../utils/constants";
```
Comment (erenfn) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:5]: Change this file too. You don't have these colors in constants

File: frontend/src/components/CheckIcon/CheckIcon.jsx [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+        <CheckCircleIcon sx={{ color: { color } }} />
+      )}
+    </div>
+  );
+}
+
+CheckIcon.propTypes = {
+  size: PropTypes.oneOf(["small", "medium", "large"]),
+  outline: PropTypes.oneOf([true, false]),
+  color: PropTypes.oneOf([purple, black, green]),
```
Comment (erenfn) [inline frontend/src/components/CheckIcon/CheckIcon.jsx:24]: same here

File: frontend/src/components/CreateActivityButton/CreateActivityButton.jsx [hunk 23]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import Button from "../Button/Button";
+import "./CreateActivityButtonStyles.css";
+import CheckIcon from "../CheckIcon/CheckIcon";
+import PropTypes from "prop-types";
+import { activityData, ACTIVITY_TYPES } from "../../data/CreateActivityButtonData";
+import '../../styles/variables.css'
```
Comment (erenfn) [inline frontend/src/components/CreateActivityButton/CreateActivityButton.jsx:7]: import this in the css file not here

File: frontend/src/styles/variables.css [hunk 24]
```diff
@@ -6,4 +6,6 @@
     --light-gray: #EAECF0;
     --light-border-color: #D0D5DD;
     --border-error-solid: #D92D20;
+    --checkIcon-black: #475467;
```
Comment (erenfn) [inline frontend/src/styles/variables.css:9]: We already have this color here (third text color)

File: frontend/src/components/CreateActivityButton/CreateActivityButtonStyles.css [hunk 25]
```diff
@@ -0,0 +1,39 @@
+.bannerStyle{
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 100vh;
+    flex-direction: column;
+}
+
+.heading{
+    color: #344054;
```
Comment (erenfn) [inline frontend/src/components/CreateActivityButton/CreateActivityButtonStyles.css:10]: use --main-text-color from variables.css

File: frontend/src/components/CreateActivityButton/CreateActivityButtonStyles.css [hunk 26]
```diff
@@ -0,0 +1,39 @@
...
+
+.list{
+    display: flex;
+    align-items: center;
+    gap: 15px;
+    padding: 5px;
+}
+
+.listText{
+    color: #344054;
```
Comment (erenfn) [inline frontend/src/components/CreateActivityButton/CreateActivityButtonStyles.css:24]: use --main-text-color from variables.css

**Review by erenfn — CHANGES_REQUESTED** [review]

**Review by uparkalau — DISMISSED** [review]

**Review by uparkalau — APPROVED** [review]

**Review by uparkalau — APPROVED** [review]

**Pull request merged**